crates/project_panel/src/project_panel.rs

```diff
@@ -82,6 +82,7 @@ pub struct ProjectPanel {
     show_scrollbar: bool,
     scrollbar_drag_thumb_offset: Rc<Cell<Option<f32>>>,
     hide_scrollbar_task: Option<Task<()>>,
+    diagnostic_error_paths: HashSet<PathBuf>,
```

Comment:
I think you need something like `HashMap<WorktreeId, HashSet<PathBuf>`, otherwise you'll have global diagnostic errors, even if they appear in a different worktree.

Example:
![screenshot-2024-09-24-15 00 32@2x](https://github.com/user-attachments/assets/b6d6ea08-94cf-4fee-b41c-4ced45a5aade)

Here I have an error in the rust-proj, but the go-proj is also marked red.

(cc @bennetbo who found this bug)

Reply:
I didn't know you can do this in Zed, thanks for flagging!

Reply:
Takes a while to digest that and then you see it everywhere :)